ansible/inventory.py

```diff
@@ -27,42 +28,82 @@ def populatevlans():
         for line in lines:
             if not (line[0] == '/' or line[0] == ' ' or line[0] == '\n'):
                 elems = re.split(r'\t+', line)
-                ipv6 = elems[3].split('/')
-                ipv6prefix = ipv6[0]
-                ipv6bitmask = ipv6[1]
-                ipv6dhcp = dhcp6ranges(ipv6prefix, int(ipv6bitmask))
-                ipv4 = elems[4].split('/')
-                ipv4prefix = ipv4[0]
-                ipv4bitmask = ipv4[1]
-                ipv4dhcp = dhcp4ranges(ipv4prefix, int(ipv4bitmask))
-                ipv4netmask = bitmasktonetmask(int(ipv4bitmask))
-                vlanid = elems[2]
-                if not vlanid.isdigit():
-                    continue
-                vlans.append({
-                    "name": elems[1],
-                    "id": vlanid,
-                    "ipv6prefix": ipv6prefix,
-                    "ipv6bitmask": ipv6bitmask,
-                    "ipv4prefix": ipv4prefix,
-                    "ipv4bitmask": ipv4bitmask,
-                    "building": file,
-                    "description": elems[5].rstrip(),
-                    "ipv6dhcp1a": ipv6dhcp[0],
-                    "ipv6dhcp1b": ipv6dhcp[1],
-                    "ipv6dhcp2a": ipv6dhcp[2],
-                    "ipv6dhcp2b": ipv6dhcp[3],
-                    "ipv4dhcp1a": ipv4dhcp[0],
-                    "ipv4dhcp1b": ipv4dhcp[1],
-                    "ipv4dhcp2a": ipv4dhcp[2],
-                    "ipv4dhcp2b": ipv4dhcp[3],
-                    "ipv4router": ipv4dhcp[4],
-                    "ipv4netmask": ipv4netmask,
-                    "ipv6dns1": "",
-                    "ipv6dns2": "",
-                    "ipv4dns1": "",
-                    "ipv4dns2": "",
-                })
+                directive = elems[0]
+                if directive == "VLAN":
```

Comment:
@kylerisse Should we have a fall through case? Im just wondering if an `else` would be a good idea if for some reason there was a `directive` other than "LVAN" or "VVRNG"

Reply:
Maybe even just an `else` that raises an error?

Reply:
We're only concerned with these two directives here. There's nothing to do in this context should anything else show up. The suggestion from the last paragraph in https://github.com/socallinuxexpo/scale-network/issues/149#issue-416179334 is to ignore and emit an error. I'd prefer not to plumb in a separate error logging facility at this time to reduce the number of last minute changes. I could throw it to STDERR but the user will never see it anyways since ansible is redirecting it's output to itself, at least that's my assumption as I'd have to do further testing. It's also expecting a pure json blob which could be broken throwing extra strings into the mix. The downside isn't worth the risk for this week, imo. When tackling #149 that's something that can be explored though, logging facilities and more appropriate handling of directives.

Reply:
Ahh gotcha, good points!